java/src/main/native/ai_onnxruntime_OnnxTensor.c

```diff
@@ -14,45 +14,64 @@
     const OrtApi* api = (const OrtApi*) apiHandle;
     OrtAllocator* allocator = (OrtAllocator*) allocatorHandle;
     // Convert type to ONNX C enum
     ONNXTensorElementDataType onnxType = convertToONNXDataFormat(onnxTypeJava);
 
     // Extract the shape information
-    jboolean mkCopy;
-    jlong* shapeArr = (*jniEnv)->GetLongArrayElements(jniEnv,shape,&mkCopy);
-    jsize shapeLen = (*jniEnv)->GetArrayLength(jniEnv,shape);
+    jlong* shapeArr = (*jniEnv)->GetLongArrayElements(jniEnv, shape, NULL);
+    jsize shapeLen = (*jniEnv)->GetArrayLength(jniEnv, shape);
 
     // Create the OrtValue
-    OrtValue* ortValue;
-    checkOrtStatus(jniEnv, api, api->CreateTensorAsOrtValue(allocator,(int64_t*)shapeArr,shapeLen,onnxType,&ortValue));
-    (*jniEnv)->ReleaseLongArrayElements(jniEnv,shape,shapeArr,JNI_ABORT);
-
-    // Get a reference to the OrtValue's data
-    uint8_t* tensorData;
-    checkOrtStatus(jniEnv, api, api->GetTensorMutableData(ortValue, (void**) &tensorData));
-
-    // Check if we're copying a scalar or not
-    if (shapeLen == 0) {
-        // Scalars are passed in as a single element array
-        copyJavaToPrimitiveArray(jniEnv, onnxType, tensorData, dataObj);
-    } else {
-        // Extract the tensor shape information
-        OrtTensorTypeAndShapeInfo* info;
-        checkOrtStatus(jniEnv,api,api->GetTensorTypeAndShape(ortValue, &info));
-        size_t dimensions;
-        checkOrtStatus(jniEnv,api,api->GetDimensionsCount(info,&dimensions));
-        size_t arrSize;
-        checkOrtStatus(jniEnv,api,api->GetTensorShapeElementCount(info,&arrSize));
-        ONNXTensorElementDataType onnxTypeEnum;
-        checkOrtStatus(jniEnv,api,api->GetTensorElementType(info,&onnxTypeEnum));
-        api->ReleaseTensorTypeAndShapeInfo(info);
-
-        // Copy the java array into the tensor
-        copyJavaToTensor(jniEnv, onnxType, tensorData, arrSize, dimensions, dataObj);
+    OrtValue* ortValue = NULL;
+    OrtErrorCode code = checkOrtStatus(jniEnv, api,
+                                       api->CreateTensorAsOrtValue(
+                                           allocator, (int64_t*)shapeArr, shapeLen, onnxType, &ortValue
+                                           )
+                                       );
+    (*jniEnv)->ReleaseLongArrayElements(jniEnv, shape, shapeArr, JNI_ABORT);
+
+    if (code == ORT_OK) {
+      // Get a reference to the OrtValue's data
+      uint8_t* tensorData;
+      code = checkOrtStatus(jniEnv, api, api->GetTensorMutableData(ortValue, (void**)&tensorData));
+      if (code == ORT_OK) {
+        // Check if we're copying a scalar or not
+        if (shapeLen == 0) {
+          // Scalars are passed in as a single element array
+          size_t copied = copyJavaToPrimitiveArray(jniEnv, onnxType, tensorData, dataObj);
+          if (copied == 0) {
+            // Copy failed, release value
+            api->ReleaseValue(ortValue);
+            ortValue = NULL;
+          }
```

Comment:
This code seems to be repeated. Suggest to set an error code, and check for it at the end of the function. If the error code is not OK, release OrtValue.

Reply:
Will do.